utils/firestore.js

```diff
@@ -2065,7 +2065,30 @@ const getScheduledNotifications = async (scheduleAt) => {
   return notificationSpecArray;
 };
 
-const getNotification = async (id) => {
+/**
+ * Get all notification specifications that are not drafts and match the scheduleAt time. This function is run once per day, for production use.
+ * @param {string} scheduleAt Time of day to send notifications, eg. '15:00'
+ * @returns 
+ */
+const getNotificationSpecsByScheduleOncePerDay = async (scheduleAt) => {
+  const currTime = new Date().toISOString();
+  const batch = db.batch();
+  const snapshot = await db.collection("notificationSpecifications").where("scheduleAt", "==", scheduleAt).get();
+  let notificationSpecArray = [];
+  for (const doc of snapshot.docs) {
+    const docData = doc.data();
+    const lastRunDate = docData.lastRunTime?.split("T")[0] ?? "";
+    if (!docData.isDraft && docData.id && currTime.split("T")[0] !== lastRunDate) {
```

Comment:
very nice

Reply:
Thanks. I just realize that we should use US east timezone instead of UTC time to check a date. Doing updates now.